#Clustering of Trichodesmium 16S sequences from Canary Island samples

##Starting point
A biom file generated by open-otu clustering of 16S reads at 97% identity using the GreenGenes_97 (v13.8) database. You will also need the seqs.fna file that was used to make this.

The following commands assume that your starting biom file is open_otus/otu_table_mc2_w_tax_no_pynast_failures.biom and your sequence file is samples/seqs.fna.

##Aim
We want to extract the subset of 16S sequences that come from Trichodesmium, and cluster these at higher levels of sequence identity (99% and above) to determine how many (sub)species we have.

##Method

###Setup

Make a new directory to carry out the analysis in.

```bash
mkdir Tricho_16S
cd Tricho_16S
```

###Extract Trichodesmium sequences from OTU table

Due to the way that the taxonomic classification works (the GG database is clustered at 97% identity, and a representative of each cluster is picked at random for use in taxonomy assignment) the Trichodesmium genus is actually labelled as Hydrocoleum.

To generate a Trichodesmium-only biom file:

```bash
filter_taxa_from_otu_table.py -i ../open_otus/otu_table_mc2_w_tax_no_pynast_failures.biom -o Tricho.biom -p g__Hydrocoleum

#To get a summary of the number of sequences in the new table:
biom summarize-table -i Tricho.biom

#To generate a text version of the otu table
biom convert -i Tricho.biom -o Tricho_otu_table.txt --to-tsv
```

Next we need to extract these same Trichodesmium sequences from the original seqs.fna file so that we can redo the clustering. (This should be easier than it is...)

First, use the text version of the OTU table to get a list of OTU IDs present in the filtered dataset

```bash
grep -v '^#' Tricho_otu_table.txt | cut -f1 > otu_list
```

Then use this list to filter the OTU mapping file (that was made when you did the original OTU picking) to keep IDs of sequences that are classed as Trichodesmium

```bash
grep -w -f otu_list ../open_otus/final_otu_map_mc2.txt > filtered_Tricho_otu_map.txt
```

Finally, use this list of sequence IDs to filter the original seqs.fna file to keep only the Trichodesmium sequences

```bash
filter_fasta.py -f ../samples/seqs.fna -o Tricho_seqs.fna -m filtered_Tricho_otu_map.txt
```

###De novo sequence clustering at 99% identity

Now we want to cluster the extracted sequences de novo at an identity level of 99%. QIIME clustering programs will default to clustering at 97%, so we need to provide a parameter file to override this behaviour. We also want to assign taxonomy using a database that is more specific than QIIME's default (GreenGenes, clustered at 97% identity). This example uses the full (unclustered) version of SILVA v111.

[TODO: add info on where to download this.]

Using a text editor, make a file containing the following and save it as de_novo_params99.txt

```bash
split_libraries_fastq:phred_offset  33
pick_otus:enable_rev_strand_match   True
pick_otus:similarity    0.99
assign_taxonomy:reference_seqs_fp   /research/miseq/db/Silva_111_post/rep_set/Silva_111_full_unique.fasta
assign_taxonomy:id_to_taxonomy_fp   /research/miseq/db/Silva_111_post/taxonomy/Silva_111_taxa_map_full.txt
```

Run de novo OTU picking

```bash
pick_de_novo_otus.py -i Tricho_seqs.fna -o denovo_otus99 -p de_novo_params99.txt
```

Clean up the result a little by removing very rare OTUs (<0.1% of the total) - these are likely to be sequencing errors.

```bash
filter_otus_from_otu_table.py -i denovo_otus99/otu_table.biom -o denovo_otus99/otu_table_filt0001.biom --min_count_fraction 0.001

biom convert -i denovo_otus99/otu_table_filt0001.biom -o denovo_otus99/otu_table_filt0001.txt --to-tsv
```

We then need to apply the same filtering to the rep set (the sequences chosen as representative of each OTU).

```bash
filter_fasta.py -f denovo_otus99/rep_set/Tricho_seqs_rep_set.fasta -o denovo_otus99/rep_set/Tricho_seqs_rep_set_filt0001.fasta -b denovo_otus99/otu_table_filt0001.biom
```

###Adding known Tricho sequences

We want to make a phylogenetic tree from the OTUs observed in our sample. However, this will be more informative if we also include some known Trichodesmium 16S sequences so that we can see where our samples fit. We can extract these sequences from the Silva database.

```bash
mkdir known_tricho_seqs

#Fetch all taxonomy entries that contain 'Trichodesmium'
grep Trichodesmium /research/miseq/db/Silva_111_post/taxonomy/Silva_111_taxa_map_full.txt > known_tricho_seqs/known_tricho_rep_set_taxSilva.txt

#Then pull the corresponding sequences out of the Silva fasta file
grep -A1 -f <(cut -f1 known_tricho_seqs/known_tricho_rep_set_taxSilva.txt) --no-group-separator /research/miseq/db/Silva_111_post/rep_set/Silva_111_full_unique.fasta > known_tricho_seqs/known_tricho_rep_set.fasta
```

Next, we create a new phylogenetic tree that includes the known sequences with our samples. Use MAFFT alignment as it seems to cope better with aligning sequences of different lengths.

```bash
mkdir align_otus_w_known_seqs

#Combine all the sequences into one file
cat denovo_otus99/rep_set/Tricho_seqs_rep_set_filt0001.fasta known_tricho_seqs/known_tricho_rep_set.fasta > align_otus_w_known_seqs/all_seqs.fasta

#Make new alignment and phylogenetic tree
align_seqs.py -i align_otus_w_known_seqs/all_seqs.fasta -o align_otus_w_known_seqs/aligned_seqs -m mafft
make_phylogeny.py -i align_otus_w_known_seqs/aligned_seqs/all_seqs_aligned.fasta
```

###Visualising data in R

To generate a phylogenetic tree showing how the clustered sequences relate to known Trichodesmium species, run [this R script - phyloseq99.R](https://github.com/ajb904/EnvGen-examples/blob/gh-pages/Tricho_16S_analysis/phyloseq99.R)

The output should look like this:

![phyloseq tree](https://github.com/ajb904/EnvGen-examples/blob/gh-pages/Tricho_16S_analysis/uclust_SILVA_treeplot.pdf)

